File: api_server/models/catalog.py
from . import database, db


class Catalog(database.Model):
    catalog_id = db.Column(db.Integer, primary_key=True)
    catalog_eternal_link = db.Column(db.String(32), unique=True)
    catalog_name = db.Column(db.String(64), unique=True)

    def __init__(self, name, eternal_link=None):
        self.catalog_name = name.replace(' ', '')
        self.catalog_eternal_link = eternal_link or self.catalog_name

File: RoseliaCore/Database/Models/Catalog.fs
namespace RoseliaCore.Database.Models

open System.ComponentModel.DataAnnotations
open System.ComponentModel.DataAnnotations.Schema

[<CLIMutable>]
[<Table("catalog")>]
type Catalog = {
    [<Key; Column("catalog_id")>]
    CatalogId: int
    
    [<Column("catalog_eternal_link")>]
    CatalogEternalLink: string
    
    [<Column("catalog_name")>]
    CatalogName: string
}
